### PR TITLE
BUG: Series.resample fails on NaT index

### LINKED ISSUE #39227: BUG: Series.resample fails on NaT PeriodIndex
Failure is only on Series and only when the index is entirely `NaT`.

```
freq = "D"

df = DataFrame([1, 2], index=PeriodIndex([NaT, NaT], freq=freq))
print(df.resample(freq).sum())

ser = Series([1, 2], index=PeriodIndex([NaT, NaT], freq=freq))
print(ser.resample(freq).sum())
```

Second raises, first returns the expected

```
Empty DataFrame
Columns: []
Index: []
```

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #39227
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

This will also resolve the core issue of https://github.com/pandas-dev/pandas/pull/34998#discussion_r556463043. cc @jorisvandenbossche 

The added test:

```
    s = series.copy()
    s.index = PeriodIndex([NaT] * len(s), freq=freq)
    result = getattr(s.resample(freq), resample_method)()
```

currently fails for `freq="M"`, where upsample is used instead of downsample leading to failure since the index is not unique. I can't tell if this is a bug or an operation that should raise. It also fails for `DataFrame`, so I think is unrelated to the issue here.

cc @jbrockmendel

### CODE REVIEW

File: doc/source/whatsnew/v1.3.0.rst [hunk 1]
```diff
@@ -339,6 +339,7 @@ Groupby/resample/rolling
 - Fixed bug in :meth:`DataFrameGroupBy.sum` and :meth:`SeriesGroupBy.sum` causing loss of precision through using Kahan summation (:issue:`38778`)
 - Fixed bug in :meth:`DataFrameGroupBy.cumsum`, :meth:`SeriesGroupBy.cumsum`, :meth:`DataFrameGroupBy.mean` and :meth:`SeriesGroupBy.mean` causing loss of precision through using Kahan summation (:issue:`38934`)
 - Bug in :meth:`.Resampler.aggregate` and :meth:`DataFrame.transform` raising ``TypeError`` instead of ``SpecificationError`` when missing keys had mixed dtypes (:issue:`39025`)
+- Bug in :meth:`Series.resample` would raise when index consisted of ``NaT`` (:issue:`39227`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.3.0.rst:342]: Probably mention that this is specific to `PeriodIndex`?

Construction with a `DatetimeIndex` in your original example raises before `resampling` so the bug fix shouldnt apply to it

```
In [9]: ser = Series([1, 2], index=DatetimeIndex([NaT, NaT], freq=freq))

ValueError: Neither `start` nor `end` can be NaT

The above exception was the direct cause of the following exception:

ValueError: Inferred frequency None from passed values does not conform to passed frequency D
```

Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst:342]: Thanks!

File: pandas/core/resample.py [hunk 2]
```diff
@@ -1653,10 +1654,17 @@ def _get_period_bins(self, ax: PeriodIndex):
...
-            binner = labels = PeriodIndex(data=[], freq=self.freq, name=ax.name)
-            return binner, [], labels
+            if len(ax) == 0:
+                # if index is empty, return empty bins
+                data = []
+                bins = np.array([], dtype=np.int64)
+            else:
+                # if index is all NaT, return a single bin
+                data = [NaT]
+                bins = np.array([len(ax)])
```
Comment (jreback) [inline pandas/core/resample.py:1665]: wait why do we have NaT in an index? e.g. why don't we have 0 bins?

Comment (rhshadrach) [reply pandas/core/resample.py:1665]: Below this is the logic for when there are `NaT` along with non-`NaT` (lines 1712-1718 here):

```
        if nat_count > 0:
            # NaT handling as in pandas._lib.lib.generate_bins_dt64()
            # shift bins by the number of NaT
            bins += nat_count
            bins = np.insert(bins, 0, nat_count)
            binner = binner.insert(0, NaT)
            labels = labels.insert(0, NaT)
```

The added logic agrees with this in the case there are all NaT.

Comment (jreback) [reply pandas/core/resample.py:1665]: i c. can you unify this logic (maybe to a function)?

**Comment by jreback** [comment]
thanks @rhshadrach

**Pull request merged**